adhocteam/Head-Start-TTADP
# Monitoring: Transaction Naming Pt2

## Description of change
While looking around in new relic I noticed some transactions I was interested in were grouped incorrectly.
This adds some more custom transaction names, and refactors the naming middleware I wrote for the widget transactions in https://github.com/adhocteam/Head-Start-TTADP/pull/379.


## How to test
Check out the [transactions page](https://one.newrelic.com/launcher/nr1-core.explorer/?platform[accountId]=3062577&platform[filters]=Iihkb21haW4gPSAnQVBNJyBBTkQgdHlwZSA9ICdBUFBMSUNBVElPTicpIg==&platform[timeRange][duration]=1800000&pane=eyJuZXJkbGV0SWQiOiJhcG0tbmVyZGxldHMudHJhbnNhY3Rpb25zIiwiZW50aXR5R3VpZCI6Ik16QTJNalUzTjN4QlVFMThRVkJRVEVsRFFWUkpUMDU4T0RZM016UTJOems1In0=&sidebars[0]=eyJuZXJkbGV0SWQiOiJucjEtY29yZS5hY3Rpb25zIiwic2VsZWN0ZWROZXJkbGV0Ijp7Im5lcmRsZXRJZCI6ImFwbS1uZXJkbGV0cy50cmFuc2FjdGlvbnMifSwiZW50aXR5R3VpZCI6Ik16QTJNalUzTjN4QlVFMThRVkJRVEVsRFFWUkpUMDU4T0RZM016UTJOems1In0=&state=661f3804-856b-58fc-8a0f-64c34a5cc08f) for sandbox in new relic.

Previously downloading reports (`GET /api/activity-reports/download-all`), getting a specific report `GET /api/activity-reports/:activityReportId`), and getting report alerts section (`GET /api/activity-reports/alerts`) would be all grouped under the `GET /api/activity-reports` transaction. Now these transactions are broken out to their own names. Transactions to get a specific report by id is under `GET /api/activity-reports/*` where `*` is the stand-in for a specific id. We want to group these together regardless of the report that is requested.



<img width="402" alt="Screen Shot 2021-09-17 at 3 01 57 PM" src="https://user-images.githubusercontent.com/9893700/133846932-ba192605-1cdf-4339-b7bb-7347f49a486e.png">

## Issue(s)

None


## Checklists

### Every PR


- [n/a] Meets issue criteria
- [n/a] JIRA ticket status updated
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy


### After merge/deploy
